src/auto-evo/AutoEvoExploringTool.cs

```diff
@@ -0,0 +1,640 @@
+        autoEvoConfiguration.BiodiversityNearbyPatchIsFreePopulation = biodiversitySplitIsMutatedCheckBox.Pressed;
+        autoEvoConfiguration.BiodiversitySplitIsMutated = biodiversityNearbyPatchIsFreePopulationCheckBox.Pressed;
+        autoEvoConfiguration.LowBiodiversityLimit = (int)lowBiodiversityLimitSpinBox.Value;
+        autoEvoConfiguration.MaximumSpeciesInPatch = (int)maximumSpeciesInPatchSpinBox.Value;
+        autoEvoConfiguration.MoveAttemptsPerSpecies = (int)moveAttemptsPerSpeciesSpinBox.Value;
+        autoEvoConfiguration.MutationsPerSpecies = (int)mutationsPerSpeciesSpinBox.Value;
+        autoEvoConfiguration.NewBiodiversityIncreasingSpeciesPopulation =
+            (int)newBiodiversityIncreasingSpeciesPopulationSpinBox.Value;
+        autoEvoConfiguration.ProtectMigrationsFromSpeciesCap = protectMigrationsFromSpeciesCapCheckBox.Pressed;
+        autoEvoConfiguration.ProtectNewCellsFromSpeciesCap = protectNewCellsFromSpeciesCapCheckBox.Pressed;
+        autoEvoConfiguration.RefundMigrationsInExtinctions = refundMigrationsInExtinctionsCheckBox.Pressed;
+        autoEvoConfiguration.StrictNicheCompetition = strictNicheCompetitionCheckBox.Pressed;
+        autoEvoConfiguration.SpeciesSplitByMutationThresholdPopulationAmount =
+            (int)speciesSplitByMutationThresholdPopulationAmountSpinBox.Value;
+        autoEvoConfiguration.SpeciesSplitByMutationThresholdPopulationFraction =
+            (float)speciesSplitByMutationThresholdPopulationFractionSpinBox.Value;
+        autoEvoConfiguration.UseBiodiversityForceSplit = useBiodiversityForceSplitCheckBox.Pressed;
+    }
+
+    /// <summary>
+    ///   Run a new generation or finish the current generation async
+    /// </summary>
+    private void OnFinishOneGenerationButtonPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+
+        if (autoEvoRun?.Aborted != false || autoEvoRun.Finished)
+        {
+            // If the previous one has finished / failed
+            autoEvoRun = new AutoEvoRun(gameProperties.GameWorld, autoEvoConfiguration) { FullSpeed = true };
+            autoEvoRun.Start();
+        }
+        else
+        {
+            // If the previous one is paused
+            autoEvoRun.FullSpeed = true;
+            autoEvoRun.Continue();
+        }
+
+        // Disable these buttons
+        finishOneGenerationButton.Disabled = true;
+        runOneStepButton.Disabled = true;
+        abortButton.Disabled = false;
+    }
+
+    /// <summary>
+    ///   Apply auto-evo effects and set up related controls
+    /// </summary>
+    private void ApplyAutoEvoRun()
+    {
+        // Apply the results
+        autoEvoRun!.ApplyAllEffects(true);
+
+        // Add run results
+        RunResults results = autoEvoRun.Results!;
+        evolutionaryTree.UpdateEvolutionaryTreeWithRunResults(results, ++currentGeneration);
+        runResultsList.Add(results.MakeSummary(gameProperties.GameWorld.Map, true));
```

Comment:
Calling MakeSummary after applying auto-evo effects doesn't result in the correct summary text.